blocky process panics if the conditiononal upstream resolver receives a response without a question section

Have been using blocky on OpenWrt running on Raspberry Pi 4 board and have been noticing that the process panics when resolving a domain in conditional resolver list. init.d takes care of restarting the process, but there is a brief DNS outage when that happens

Below is the stack trace. Unfortunately it is random so don't have steps to reproduce the error. 
```
panic: runtime error: index out of range [0] with length 0

goroutine 142 [running]:
github.com/0xERR0R/blocky/resolver.(*ConditionalUpstreamResolver).internalResolve(0x9a5a60?, {0xcbd510?, 0x40000a5020}, {0x4000136168, 0x10}, {0x4000136171, 0x7},

      /home/runner/work/blocky/blocky/resolver/conditional_upstream_resolver.go:111 +0x368
github.com/0xERR0R/blocky/resolver.(*ConditionalUpstreamResolver).processRequest(0x4000031260, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/conditional_upstream_resolver.go:63 +0x18c
github.com/0xERR0R/blocky/resolver.(*ConditionalUpstreamResolver).Resolve(0x4000031260, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/conditional_upstream_resolver.go:88 +0x70
github.com/0xERR0R/blocky/resolver.(*CachingResolver).Resolve(0x40000718c0, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/caching_resolver.go:189 +0x3b4
github.com/0xERR0R/blocky/resolver.(*BlockingResolver).Resolve(0x40004ac600, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/blocking_resolver.go:399 +0x98
github.com/0xERR0R/blocky/resolver.(*HostsFileResolver).Resolve(0x400009ce70?, 0x400018d7f8?)
      /home/runner/work/blocky/blocky/resolver/hosts_file_resolver.go:106 +0x3f4
github.com/0xERR0R/blocky/resolver.(*CustomDNSResolver).Resolve(0x4000033b40, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/custom_dns_resolver.go:143 +0x174
github.com/0xERR0R/blocky/resolver.(*MetricsResolver).Resolve(0x400008ccd0, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/metrics_resolver.go:29 +0x38
github.com/0xERR0R/blocky/resolver.(*QueryLoggingResolver).Resolve(0x4000033b00, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/query_logging_resolver.go:113 +0x6c
github.com/0xERR0R/blocky/resolver.(*EdeResolver).Resolve(0x400009ccb0?, 0x4000273b18?)
      /home/runner/work/blocky/blocky/resolver/ede_resolver.go:24 +0x88
github.com/0xERR0R/blocky/resolver.(*ClientNamesResolver).Resolve(0x400008cc30, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/client_names_resolver.go:65 +0x150
github.com/0xERR0R/blocky/resolver.(*FqdnOnlyResolver).Resolve(0x40000312c0, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/fqdn_only_resolver.go:36 +0x108
github.com/0xERR0R/blocky/resolver.(*FilteringResolver).Resolve(0x4000031290, 0x400009cd20)
      /home/runner/work/blocky/blocky/resolver/filtering_resolver.go:33 +0x1b0
github.com/0xERR0R/blocky/server.(*Server).OnRequest(0x40002100f0, {0xcc4620, 0x40004ac380}, 0x4000532480)
      /home/runner/work/blocky/blocky/server/server.go:610 +0xc8
github.com/miekg/dns.HandlerFunc.ServeDNS(0x40000a4060?, {0xcc4620?, 0x40004ac380?}, 0x4000530001?)
      /home/runner/go/pkg/mod/github.com/miekg/dns@v1.1.52/server.go:37 +0x38
github.com/miekg/dns.(*ServeMux).ServeDNS(0x400032a000?, {0xcc4620?, 0x40004ac380?}, 0x4000532480?)
      /home/runner/go/pkg/mod/github.com/miekg/dns@v1.1.52/serve_mux.go:103 +0x78
github.com/miekg/dns.(*Server).serveDNS(0x4000096120, {0x400032a000, 0x22, 0xffff}, 0x40004ac380)
      /home/runner/go/pkg/mod/github.com/miekg/dns@v1.1.52/server.go:659 +0x3a0
github.com/miekg/dns.(*Server).serveUDPPacket(0x4000096120, 0x28dc5c?, {0x400032a000, 0x22, 0xffff}, {0xcc2400?, 0x400011a038}, 0x40004c0d20, {0x0?, 0x0})
      /home/runner/go/pkg/mod/github.com/miekg/dns@v1.1.52/server.go:603 +0x1b8
created by github.com/miekg/dns.(*Server).serveUDP
      /home/runner/go/pkg/mod/github.com/miekg/dns@v1.1.52/server.go:533 +0x3e0
```

Blocky version. Running the arm64 binary available in github releases.
```
blocky
Version: v0.21
Build time: 20230327-054311
Architecture: undefined
```

OpenWRT version
```
NAME="OpenWrt"
VERSION="22.03.5"
ID="openwrt"
ID_LIKE="lede openwrt"
PRETTY_NAME="OpenWrt 22.03.5"
VERSION_ID="22.03.5"
HOME_URL="https://openwrt.org/"
BUG_URL="https://bugs.openwrt.org/"
SUPPORT_URL="https://forum.openwrt.org/"
BUILD_ID="r20134-5f15225c1e"
OPENWRT_BOARD="bcm27xx/bcm2711"
OPENWRT_ARCH="aarch64_cortex-a72"
OPENWRT_TAINTS=""
OPENWRT_DEVICE_MANUFACTURER="OpenWrt"
OPENWRT_DEVICE_MANUFACTURER_URL="https://openwrt.org/"
OPENWRT_DEVICE_PRODUCT="Generic"
OPENWRT_DEVICE_REVISION="v0"
OPENWRT_RELEASE="OpenWrt 22.03.5 r20134-5f15225c1e"
```

Conditional upstream config is fairly basic.

```yml
conditional:
  mapping:
    jio.com: '192.168.10.1, 192.168.11.1, 2405:200:800::3, 49.45.0.3'
    jiokhelo.com: '192.168.10.1, 192.168.11.1, 2405:200:800::3, 49.45.0.3'
    jfdca.net: '192.168.10.1, 192.168.11.1, 2405:200:800::3, 49.45.0.3'
    jiocinema.com: '192.168.10.1, 192.168.11.1, 2405:200:800::3, 49.45.0.3'
    voot.com: '192.168.10.1, 192.168.11.1, 2405:200:800::3, 49.45.0.3'
```

--- a/resolver/conditional_upstream_resolver.go
+++ b/resolver/conditional_upstream_resolver.go
@@ -108,7 +108,10 @@ func (r *ConditionalUpstreamResolver) internalResolve(reso Resolver, doFQ, do st
 	if err == nil {
 		response.Reason = "CONDITIONAL"
 		response.RType = model.ResponseTypeCONDITIONAL
-		response.Res.Question[0].Name = req.Req.Question[0].Name
+
+		if len(response.Res.Question) > 0 {
+			response.Res.Question[0].Name = req.Req.Question[0].Name
+		}
 	}
 
 	var answer string

--- a/resolver/conditional_upstream_resolver_test.go
+++ b/resolver/conditional_upstream_resolver_test.go
@@ -47,12 +47,23 @@ var _ = Describe("ConditionalUpstreamResolver", Label("conditionalResolver"), fu
 		})
 		DeferCleanup(dotTestUpstream.Close)
 
+		refuseTestUpstream := NewMockUDPUpstreamServer().WithAnswerFn(func(request *dns.Msg) (response *dns.Msg) {
+			response = new(dns.Msg)
+			response.Rcode = dns.RcodeRefused
+			// question section in response should be empty
+			request.Question = make([]dns.Question, 0)
+
+			return response
+		})
+		DeferCleanup(refuseTestUpstream.Close)
+
 		sut, _ = NewConditionalUpstreamResolver(config.ConditionalUpstreamConfig{
 			Mapping: config.ConditionalUpstreamMapping{
 				Upstreams: map[string][]config.Upstream{
-					"fritz.box": {fbTestUpstream.Start()},
-					"other.box": {otherTestUpstream.Start()},
-					".":         {dotTestUpstream.Start()},
+					"fritz.box":      {fbTestUpstream.Start()},
+					"other.box":      {otherTestUpstream.Start()},
+					"refused.domain": {refuseTestUpstream.Start()},
+					".":              {dotTestUpstream.Start()},
 				},
 			},
 		}, nil, false)
@@ -78,6 +89,21 @@ var _ = Describe("ConditionalUpstreamResolver", Label("conditionalResolver"), fu
 	})
 
 	Describe("Resolve conditional DNS queries via defined DNS server", func() {
+		When("conditional resolver returns error code", func() {
+			It("Should be returned without changes", func() {
+				Expect(sut.Resolve(newRequest("refused.domain.", A))).
+					Should(
+						SatisfyAll(
+							HaveNoAnswer(),
+							HaveResponseType(ResponseTypeCONDITIONAL),
+							HaveReason("CONDITIONAL"),
+							HaveReturnCode(dns.RcodeRefused),
+						))
+
+				// no call to next resolver
+				Expect(m.Calls).Should(BeEmpty())
+			})
+		})
 		When("Query is exact equal defined condition in mapping", func() {
 			Context("first mapping entry", func() {
 				It("Should resolve the IP of conditional DNS", func() {

